### DISCUSSION TITLE
Note-FastLED - how to change the order of LEDs?

### DISCUSSION THREAD

**Discussion by Kmanggaard** [tttapa/Control-Surface#730] [Q&A]
The example works great! Since my DAW sends midi values after Novation standard the colours match nicely "out of the box". Unfortunately, my physical LEDs (on PCB) are responding to control messages ordered:

51, 54, 58, 62, 66, 70, 74, 78,
52, 55, 59, 63, 67, 71, 75, 79,
53, 56, 60, 64, 68, 72, 76, 80,
54, 57, 61, 65, 69, 73, 77, 81,

...and then eight non-sequential CCs.

Is there a "CC-FastLED"?
And how can I change the order of the LEDs? 

Regards and respect.

**Comment by tttapa** [comment]
> Is there a "CC-FastLED"?

Yes, see [`CCRangeFastLED`](https://tttapa.github.io/Control-Surface-doc/Doxygen/d7/de4/group__midi-input-elements-leds.html#gaa7812b2ef3dc8db4c246c9a452db81b4).

> And how can I change the order of the LEDs?

You can use the [`setLEDIndexPermuter`](https://tttapa.github.io/Control-Surface-doc/Doxygen/d3/d98/classNoteCCKPRangeFastLED.html#a353b91dbf60c992aa7de7861136f339d) function.

**Reply by Kmanggaard** [reply]
Thank you very much. I obviously didn't look hard enough. Your hard work is much appreciated.

**Reply by tttapa** [reply]
No worries, the documentation can be hard to navigate sometimes.
